v3: Plugins that update service-data generate inconsistent code

Using the plugin that affect service-data, the code generated by `goa gen` and `goa example` may become inconsistent.

## Example design

```go
package design

import (
   . "goa.design/goa/v3/dsl"
   cors "goa.design/plugins/v3/cors/dsl"
)

var _ = API("calc", func() {
   Title("Calculator Service")
   Description("Service for adding numbers, a Goa teaser")
   Server("calc", func() {
      Host("localhost", func() {
         URI("http://localhost:8000")
         URI("grpc://localhost:8080")
      })
   })
   cors.Origin("/.*localhost.*/", func() {
      cors.Headers("Content-Type", "api_key", "Authorization")
      cors.Methods("GET", "POST", "DELETE", "PUT", "PATCH", "OPTIONS")
      cors.MaxAge(600)
   })
})

var _ = Service("calc", func(){
   Files("/openapi.json", "./gen/http/openapi.json")
})
```

## goa gen & goa example

It can not be compiled because the arguments are different between the example server and the code generated by `goa gen`. 

```shellsession
$ goa gen calc/design
...snip
$ goa example calc/design
...snip
$ go build ./cmd/calc
# calc/cmd/calc
cmd/calc/http.go:58:15: not enough arguments in call to server.Mount
        have ("goa.design/goa/v3/http".Muxer)
        want ("goa.design/goa/v3/http".Muxer, *server.Server)
```

Example server
cmd/calc/http.go
```go
    // Configure the mux.
    calcsvr.Mount(mux)
```

Corresponding template:
goa/http/codegen/example_server.go
```go
	// Configure the mux.
	{{- range .Services }}
		{{ .Service.PkgName }}svr.Mount(mux{{ if .Endpoints }}, {{ .Service.VarName }}Server{{ end }})
	{{- end }}
```

Gen code
gen/calc/server/server.go
```go
    // Mount configures the mux to serve the calc endpoints.
    func Mount(mux goahttp.Muxer, h *Server) {
       MountCORSHandler(mux, h.CORS)
       MountGenHTTPOpenapiJSON(mux, http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
          http.ServeFile(w, r, "./gen/http/openapi.json")
       }))
    }
```

Corresponding template:
goa/http/codegen/server.go
```go
// input: ServiceData
const serverMountT = `{{ printf "%s configures the mux to serve the %s endpoints." .MountServer .Service.Name | comment }}
func {{ .MountServer }}(mux goahttp.Muxer{{ if .Endpoints }}, h *{{ .ServerStruct }}{{ end }}) {
	{{- range .Endpoints }}
	{{ .MountHandler }}(mux, h.{{ .Method.VarName }})
	{{- end }}
```

## Bug ?

Perhaps I think this causes because the plugins does not affect the creation of the example server. CORS plugin affects in the case of `goa gen`, but not in the case of `goa example`.

cors/generate.go
```go
    // serverCORS updates the HTTP server file to handle preflight paths and
    // adds the required CORS headers to the response.
    func serverCORS(f *codegen.File) {
       if filepath.Base(f.Path) != "server.go" {
          return
       }
    
       var svcData *ServiceData
       for _, s := range f.Section("server-struct") {
    
          data, ok := s.Data.(*httpcodegen.ServiceData)
          if !ok { // other transport, e.g. gRPC
             continue
          }
    
          codegen.AddImport(f.SectionTemplates[0],
             &codegen.ImportSpec{Path: "goa.design/plugins/v3/cors"})
    
          if d, ok := ServicesData[data.Service.Name]; !ok {
             svcData = buildServiceData(data.Service.Name)
             ServicesData[data.Service.Name] = svcData
          } else {
             svcData = d
          }
          for _, o := range svcData.Origins {
             if o.Regexp {
                codegen.AddImport(f.SectionTemplates[0],
                   &codegen.ImportSpec{Path: "regexp"})
                break
             }
          }
          data.Endpoints = append(data.Endpoints, svcData.Endpoint) // ← ★ Add endpoint
          fm := codegen.TemplateFuncs()
          f.SectionTemplates = append(f.SectionTemplates, &codegen.SectionTemplate{
```

It is a bug. It happens only if the cors plugin is used in a Service that contains only a file server.
```
cmd/calc/http.go:58:15: not enough arguments in call to server.Mount
        have ("goa.design/goa/v3/http".Muxer)
        want ("goa.design/goa/v3/http".Muxer, *server.Server)
```
But if the Service defines other endpoints, we won't hit this bug. I think the plugin should be aware of this scenario and generate the appropriate example code.

--- a/cors/generate.go
+++ b/cors/generate.go
@@ -2,6 +2,7 @@ package cors
 
 import (
 	"path/filepath"
+	"regexp"
 	"strings"
 
 	"goa.design/goa/v3/codegen"
@@ -33,6 +34,7 @@ type (
 // Register the plugin Generator functions.
 func init() {
 	codegen.RegisterPlugin("cors", "gen", nil, Generate)
+	codegen.RegisterPlugin("cors-example", "example", nil, TweakExample)
 }
 
 // Generate produces server code that handle preflight requests and updates
@@ -40,6 +42,24 @@ func init() {
 func Generate(genpkg string, roots []eval.Root, files []*codegen.File) ([]*codegen.File, error) {
 	for _, f := range files {
 		serverCORS(f)
+	}
+	return files, nil
+}
+
+// TweakExample handles the special case where a service only has file servers
+// and no method in which case the Goa generator generate a Mount method that
+// does not take a second argument but this plugin generates one that does. The
+// second argument is the actual HTTP server which is needed so it can be
+// configured with the CORS endpoint. So this method simply removes the special
+// case from the Goa template generating the example.
+func TweakExample(genpkg string, roots []eval.Root, files []*codegen.File) ([]*codegen.File, error) {
+	re := regexp.MustCompile("{{ if .Endpoints }}(.+){{ end }}")
+	for _, f := range files {
+		for _, t := range f.SectionTemplates {
+			if t.Name == "server-http-init" {
+				t.Source = re.ReplaceAllString(t.Source, "$1")
+			}
+		}
 	}
 	return files, nil
 }

